api/forgot.go

```diff
@@ -96,9 +107,8 @@ func (a *Api) passwordReset(res http.ResponseWriter, req *http.Request, vars map
 		resetCnf.UpdateStatus(models.StatusCompleted)
 	}
 
-	if resetCnf != nil && a.addOrUpdateConfirmation(resetCnf, res) {
+	if resetCnf != nil && (info != nil || a.addOrUpdateConfirmation(resetCnf, res)) {
```

Comment:
so this means we do not consider the failure of storing the confirmation in the DB when we deal with an info email? I do not understand why

Reply:
This one is used to save confirmation if info is nil. In the case info is not nil the addOrUpdateConfirmation will not be called and so the confirmation will not be saved in the database.

Reply:
ok got it